cmake/Modules/FindComputeCpp.cmake

```diff
@@ -261,54 +263,83 @@ function(__build_spir targetName sourceFile binaryDir fileCounter)
     set(device_compiler_cxx_standard "")
   endif()
 
+  get_property(source_compile_flags
+    SOURCE ${SDK_BUILD_IR_SOURCE}
+    PROPERTY COMPUTECPP_SOURCE_FLAGS
+  )
+  if(source_compile_flags)
+    list(APPEND target_compile_flags ${source_compile_flags})
+  endif()
+
   set(COMPUTECPP_DEVICE_COMPILER_FLAGS
     ${device_compiler_cxx_standard}
     ${COMPUTECPP_DEVICE_COMPILER_FLAGS}
-    ${COMPUTECPP_USER_FLAGS})
-  # Convert argument list format
+    ${COMPUTECPP_USER_FLAGS}
+    ${target_compile_flags}
+  )
   separate_arguments(COMPUTECPP_DEVICE_COMPILER_FLAGS)
 
+  set(ir_dependencies ${SDK_BUILD_IR_SOURCE})
+  if(target_libraries)
```

Comment:
Somewhere in your changes you removed the code that would set this variable. This could mean that library dependencies aren't handled correctly in the custom command.